src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -79,7 +80,14 @@ public CommandResult execute(Model model) throws CommandException {
         Person personToEdit = lastShownList.get(index.getZeroBased());
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+        // A person's uniqueness is based on their phone and email. If the edit command results in sharing those
+        // identifiers with 2 others, it means the uniqueness invariant was violated and hence there are duplicates.
+        if (countDuplicatePerson(fullPersonList, editedPerson) == 2) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        }
+        // This covers an edge case, when the edit command changes both the phone and email to fully match another
+        // existing person, which will circumvent the above check.
```

Comment:
This is not strictly true, I could change the email that match another existing person, and change the phone to a brand new number. 

In this case, the new person does not fully match another existing person, yet it circumvent the above check.

Reply:
I have changed the comment.